### PR TITLE
fix: solve issues around OpenAPI query param definitions which are ca…

### PR DESCRIPTION
**Pull request by claudemamo**
…using code generation to choke

### CODE REVIEW

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/openapi/Property.java [hunk 1]
```diff
@@ -59,6 +60,7 @@
  */
 @Value
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
+@Getter
```
Comment (jbee) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/openapi/Property.java:63]: `@Value` does imply `@Getter` I think

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/openapi/Property.java [hunk 2]
```diff
@@ -101,7 +110,19 @@ private static boolean isProperty(Field source) {
     return !isExcluded(source);
   }
 
-  private static boolean isProperty(Method source) {
+  private static boolean isSetter(Method source) {
+    String name = source.getName();
+    return !isExcluded(source)
+        && source.getParameterCount() == 1
+        && Stream.of("set")
```
Comment (jbee) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/openapi/Property.java:117]: This seems unnecessarily complicated. 
Just `name.startsWith("set") && name.length() > 3 && isUpperCase(name.charAt(3))`  seems more clear.

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/openapi/Property.java [hunk 3]
```diff
@@ -75,7 +77,11 @@ private Property(Field f) {
   }
 
   private Property(Method m) {
-    this(getName(m), getType(m, m.getGenericReturnType()), m, isRequired(m, m.getReturnType()));
+    this(
+        getName(m),
+        getType(m, isSetter(m) ? m.getParameterTypes()[0] : m.getGenericReturnType()),
```
Comment (jbee) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/openapi/Property.java:82]: I think this should be `m.getGenericParameterTypes()[0]`

File: dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/TrackerIdSchemeParam.java [hunk 4]
```diff
@@ -45,6 +45,9 @@
 @Builder
 @JsonDeserialize(builder = TrackerIdSchemeParam.TrackerIdSchemeParamBuilder.class)
 @AllArgsConstructor(staticName = "of", access = AccessLevel.PRIVATE)
+// FIXME: omitting this will result in a complex type for the query param in the OpenAPI definition
```
Comment (jbee) [inline dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/TrackerIdSchemeParam.java:48]: I think we should leave this as `TODO` not `FIXME`.

**Review by jbee — CHANGES_REQUESTED** [review]

**Review by jbee — APPROVED** [review]

**Review by david-mackessy — APPROVED** [review]

**Review by netroms — APPROVED** [review]

**Pull request merged**